kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -368,6 +373,20 @@ public boolean isBlindAppend() {
     return true;
   }
 
+  private static PostCommitAction checkpoint(Engine engine, String tablePath, long version) {
```

Comment:
I'd prefer a different organization here for post commit hooks. we don't know how many there will be, and how complex their implementations will be.

let's create a new package `/internal/hooks` and create `class PostCommitCheckpoint implements PostCommitHook`

another alterinative is `/internals/hooks/PostCommitHooks.java` which has nested static classes. I prefer the former, open to your opinion, too.

Reply:
Updated the implementation following how spark does, put all under /internal/hooks and implement a class called checkpointHook

https://github.com/delta-io/delta/tree/a6db4e05d943fec019522c9c08ff0b64125cbc5d/spark/src/main/scala/org/apache/spark/sql/delta/hooks